### PR TITLE
RA-128

### PR DESCRIPTION
**Pull request by dfernandezramos**

### CODE REVIEW

File: VAS.Services/Controller/PlaylistController.cs [hunk 1]
```diff
@@ -32,7 +32,12 @@ public class PlaylistController : DisposableBase, IController
 	{
 		PlaylistCollectionVM viewModel;
 
-		public PlaylistController (IPlayerViewModel playerVM)
+		public IVideoPlayerViewModel PlayerVM {
+			get;
+			set;
+		}
+
```
Comment (vguzmanp) [inline VAS.Services/Controller/PlaylistController.cs]: Constructor should go before properties

File: VAS.Core/Common/Cloner.cs [hunk 2]
```diff
@@ -29,15 +30,19 @@ public static T Clone<T> (this T source, SerializationType type = SerializationT
...
 
 			Stream s = new MemoryStream ();
 
 			// Binary deserialization fails in mobile platforms because of
 			// https://bugzilla.xamarin.com/show_bug.cgi?id=37300
-			#if OSTYPE_ANDROID || OSTYPE_IOS
+#if OSTYPE_ANDROID || OSTYPE_IOS
 			type = SerializationType.Json;
-			#endif
+#endif
```
Comment (vguzmanp) [inline VAS.Core/Common/Cloner.cs]: I hate when xamarin does this...

File: VAS.Core/Common/Cloner.cs [hunk 3]
```diff
@@ -29,15 +30,19 @@ public static T Clone<T> (this T source, SerializationType type = SerializationT
...
+
+			if (source is BindableBase) {
+				type = SerializationType.Json;
```
Comment (vguzmanp) [inline VAS.Core/Common/Cloner.cs]: Most of our objects are BindableBase (or will be). What does this imply?

File: VAS.Core/Store/Templates/Team.cs [hunk 4]
```diff
@@ -98,18 +97,8 @@ protected virtual void Dispose (bool disposing)
...
-				if (list != null) {
-					list.CollectionChanged -= PlayersChanged;
-				}
-				list = value;
-				if (list != null) {
-					list.CollectionChanged += PlayersChanged;
-				}
-			}
+			get;
+			set;
```
Comment (vguzmanp) [inline VAS.Core/Store/Templates/Team.cs]: Now PlayersChanged is never called. Shouldn't we call it on `List.CollectionChanged`?

File: VAS.Core/Common/Cloner.cs (hunk 3 above)
Comment (aameijide) [inline VAS.Core/Common/Cloner.cs]: This solve problems with cloned items that comes from DB. It indicates that the serialization type must be Json, like the one we have in DB.

We we're using that has a parameter in HandleSelectionChange from ProjectsController or TemplatesController, so this made that we don't have to put this parameters each time we have a Bindable base

Comment (vguzmanp) [inline VAS.Core/Common/Cloner.cs]: So why not use this always?

File: VAS.Core/Store/Templates/Team.cs (hunk 4 above)
Comment (aameijide) [inline VAS.Core/Store/Templates/Team.cs]: It's no longer needed, so I remove it

**Pull request merged**